host/traceh_enclave.c

```diff
@@ -18,6 +18,38 @@
 
 #include "core_u.h"
 
+#if !defined(OE_USE_BUILTIN_EDL)
+/**
+ * Declare the protoype of the following functions to avoid the
+ * missing-prototypes warning.
+ */
+oe_result_t _oe_log_init_ecall(
+    oe_enclave_t* enclave,
+    const char* enclave_path,
+    uint32_t log_level);
+
+/**
+ * Make the following ECALL weak to support the system EDL opt-in.
+ * When the user does not opt in (import) the EDL, the linker will picked
+ * the following default implementation. If the user opts in the EDL,
+ * the implementions (which also weak) in the oeedger8r-generated code will be
```

Comment:
Just note that we currently plan to make all the ecalls weak in order to realize the design described by #3086. See the PoC in https://github.com/openenclave/oeedger8r-cpp/pull/24.